lib/swaps/Swaps.ts

```diff
@@ -141,207 +132,218 @@ class Swaps extends EventEmitter {
+      this.logger.error(msg);
+      return msg;
+    }
+
+    // If I'm the maker I need to forward the payment to the other chain
+    // TODO: check that I got the right amount before sending out the agreed amount
+    // TODO: calculate CLTV
+    if (deal.myRole === SwapDealRole.Maker) {
+	  this.logger.debug('Executing maker code');
+      let cmdLnd = this.lndLtcClient;
+
+      switch (deal.makerCurrency) {
+        case 'BTC':
+          break;
+        case 'LTC':
+          cmdLnd = this.lndBtcClient;
+          break;
+      }
+
+      const request = new lndrpc.SendRequest();
+      request.setAmt(deal.takerAmount);
+      request.setDestString(deal.takerPubKey!);
+      request.setPaymentHashString(deal.r_hash);
+
+      try {
+        const response = await cmdLnd.sendPaymentSync(request);
+        if (response.getPaymentError()) {
+          this.logger.error('Got error from sendPaymentSync: ' + response.getPaymentError() + ' ' + JSON.stringify(request.toObject()));
+          return response.getPaymentError();
+        }
+
+        const hexString = Buffer.from(response.getPaymentPreimage_asB64(), 'base64').toString('hex');
+        this.logger.debug('got preimage ' + hexString);
+        return hexString;
+      } catch (err) {
+        this.logger.error('Got exception from sendPaymentSync: ' + ' ' + JSON.stringify(request.toObject()));
+        return 'Got exception from sendPaymentSync';
+      }
+    } else {
+      // If we are here we are the taker
+      this.logger.debug('Executing taker code');
+      if (!deal.r_preimage) {
+        this.logger.error('Do not have r_preImage. Strange.');
+        return 'Do not have r_preImage. Strange.';
+      }
+      this.logger.debug(('deal.preimage = ' + deal.r_preimage));
+      return deal.r_preimage;
+    }
+
   }
 
   /**
-   * Handles a request from a peer to create a swap deal. Creates a preimage to be used for the swap, creates a deal,
-   * and stores the deal in the local collection of deals. Responds to the peer with a deal response packet.
+   * Handles a request from a peer to create a swap deal. Creates a deal,
+   * and stores the deal in the local collection of deals. Responds to the peer with a swap response packet.
    */
-  private handleDealRequest = (requestPacket: packets.DealRequestPacket, peer: Peer)  => {
+  private handleSwapRequest = (requestPacket: packets.SwapRequestPacket, peer: Peer)  => {
     const requestBody = requestPacket.body;
     if (!requestBody) {
+      // TODO: send an error packet
       return;
     }
 
-    /** The lnd pub key to use for the local (maker) side of the deal */
-    let makerPubKey: string | undefined;
-    switch (requestBody.makerCurrency){
-      case 'BTC':
-        makerPubKey = this.pool.handshakeData.lndbtcPubKey;
-        break;
-      case 'LTC':
-        makerPubKey = this.pool.handshakeData.lndltcPubKey;
-        break;
-    }
+    let responsePacket: packets.SwapResponsePacket;
```

Comment:
Is there a reason to declare the `responsePacket` up here and not just when it's assigned a value further down?